src/ucp/rndv/rndv.c

```diff
@@ -1712,6 +1713,7 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_receive, (worker, rreq, rndv_rts_hdr, rkey_buf),
                         worker, rndv_req, rreq, rndv_rts_hdr->sreq.req_id,
                         rkey_buf, rndv_rts_hdr->address, rndv_rts_hdr->size, 0);
                 if (status == UCS_OK) {
+                    ucp_rndv_recv_data_init(rreq, rndv_rts_hdr->size);
```

Comment:
shouldn't we do it before calling ucp_rndv_recv_start_get_pipeline?
what if something completes in-place inside ucp_rndv_recv_start_get_pipeline ?